.github/workflows/run_torchbench.yml

```diff
@@ -39,10 +39,8 @@ jobs:
           # shellcheck source=/dev/null
           . "${SETUP_SCRIPT}"
           conda activate pr-ci
-          # pin cmake version to 3.22 since 3.23 breaks pytorch build
-          # see details at: https://github.com/pytorch/pytorch/issues/74985
           conda install -y numpy="${NUMPY_VERSION}" requests ninja pyyaml mkl mkl-include \
-                           setuptools cmake=3.22 cffi typing_extensions boto3 \
+                           setuptools cmake cffi typing_extensions boto3 \
```

Comment:
Our best practice is to pin the version of the dependencies (cmake) so that they can be upgraded in a controlled manner.  For this specific PR, I think you can try to up the version of cmake here to `3.23.1` or `3.23.*`.  In PyTorch CI, the version of most dependencies are now pinned and cached.  

Depending on the platforms, they are in the following locations:

* Mac, Linux, iOS conda https://github.com/pytorch/pytorch/tree/master/.github/requirements
* Default version https://github.com/pytorch/test-infra/blob/main/.github/actions/setup-miniconda/action.yml#L172
* Windows AMI conda https://github.com/pytorch/test-infra/blob/main/aws/ami/windows/scripts/Installers/Install-Conda-Dependencies.ps1#L18

These are just FYI. There is no need to update them as part of this PR.

Reply:
Thanks for the response! My current plan is to pin cmake in run_torchbench.yml to 3.23.1 within this PR.

My next plan is to make a new PR and update the cmake versions in the files you mentioned to 3.23.1 as well. Does this sound good? Is there anything I should be careful about?

Reply:
Sounds good to me.  There is no major concern about upgrading cmake version AFAIK.  It's ok as long as all the tests pass.  

Windows AMI update involves a bit of a manual testing steps (baking the new AMI and testing it), but I can help you with that.

Reply:
Just pinned the cmake version to 3.23.1 in run_torchbench.yml, I'll try testing it out --

RUN_TORCHBENCH:

Reply:
### Issue
Looking at the tests, it turns out that the default conda channel doesn't have 3.23.1. The most recent version of cmake the default channel that conda uses is 3.22.1 from [8 months ago](https://anaconda.org/anaconda/cmake), which threw me off since this person https://github.com/pytorch/pytorch/issues/74985 was able to run `conda install cmake==3.23.0`, and also ran their build on cmake version 3.23.1. I'm guessing some time between now and then, anaconda pulled those versions down, bringing us back to step one :(. I'm thinking of two possible solutions --

### Option 1: Revert
Just revert back to 3.22. Since the default conda channel uses version 3.22.1, should we standardize the cmake version here in `run_torchbench.yml` to be 3.22.1 and the places mentioned here https://github.com/pytorch/pytorch/pull/91739#discussion_r1061974102 to 3.22.1 instead of 3.22.*?

### Option 2: Use 'conda-forge' channel
Another option I was looking at was using the conda-forge channel, which has cmake [updated to version 3.25.1](https://anaconda.org/conda-forge/cmake), so the command to install it would look like: `conda install conda-forge::cmake=2.25.1` (at least for the latest version, there are previous versions we can use if 2.25.1 is too aggressive). I'm a first-time contributor so I don't know if using a different channel under conda is advisable. But from what I see, the default channel rarely updates cmake, while the conda-forge channel seems to be really on top of updates by comparison: see default vs. conda-forge --
![default-channel-cmake-versions](https://user-images.githubusercontent.com/59350561/210900856-123f9616-271b-41ec-a925-e8184b214114.png)

![conda-forge-cmake-versions](https://user-images.githubusercontent.com/59350561/210900799-1b82cd53-70b1-44f6-af20-0ef900684775.png)

What are your thoughts on which option looks best?

Reply:
IIRC, we've run into issues trying switching to conda-forge in the past. I think they had different versions of certain libraries (and in at least one case resolved dependencies to a different library altogether), which caused problems.

Going down that route might be inviting a lot of trouble.

Reply:
Oh yeah, specifically, the switch resulted in segfaults on iOS builds
More details here: https://github.com/pytorch/pytorch/issues/87208

Reply:
Yeah, we could use conda-forge if there are important features from newer version there that we need.  Otherwise, let's not bring that in.

Reply:
Makes sense, I did find it suspicious that conda-forge has multiple of the same versions of cmake like 3.20.2 or 3.18.2, maybe indicating reliability issues. So there are two things on my mind right now --

## Close relevant issues
We should close these two issues that initially brought me here: https://github.com/pytorch/pytorch/issues/74985 and https://github.com/pytorch/pytorch/issues/75705

## Reconsider use of `conda-forge` in Windows AMI?
I noticed in the file linked above by @huydhn [here](https://github.com/pytorch/test-infra/blob/main/aws/ami/windows/scripts/Installers/Install-Conda-Dependencies.ps1#L18) that the command `conda install cmake` is used on line 17, followed by `conda install -y -c conda-forge cmake=3.22.3` on line 18. I have two concerns; first is that the cmake on line 17 is unpinned, so we should pin it to `3.22` or `3.22.1` to follow the current practice of pinning cmake. The second is that it looks like there has been a determination that `conda-forge` isn't a reliable channel, and that we should only be using the main channel, so once we pin cmake to 3.22.1 on line 17, we should remove line 18. I also just realized that file is in a different repo; should I make a PR there to implement these changes?

Reply:
IMO, https://github.com/pytorch/pytorch/issues/75705 can be closed because it's outdated while https://github.com/pytorch/pytorch/issues/74985 would probably still be there till conda makes 3.23.1 available to use.

> ### Reconsider use of `conda-forge` in Windows AMI?

This makes sense to get rid of conda-forge there and pin the one from conda as you suggest, but I guess that this there for a reason (may be something broke in the past that I'm not aware of).  You can make the PR there to update the AMI. We can then do a test to see if it works with 3.22.1 like other workflows.

Reply:
> You can make the PR there to update the AMI. We can then do a test to see if it works with `3.22.*` like other workflows.

Alright, I went ahead and made one -- https://github.com/pytorch/test-infra/pull/1368

As for the tests on this PR, it looks like this is the discerning issue --
`++ git merge-base 7dd28e9e83eb74ed03682c5a4174d5c5a3711c89 90244807f701bd6ec9537d0abcca786410889cfb`
`fatal: Not a valid commit name 90244807f701bd6ec9537d0abcca786410889cfb`
I'm guessing, but it seems like TorchBench CI doesn't really like this commit existing in my forked repo of PyTorch. The only solution I can think of is cloning the master repo, creating a branch, making the changes, and opening another PR. Any ideas on resolving this so that the CI can get this commit?

Reply:
No worry, this change looks simple enough. I would be happy to merge this one for you as it stands.